components/org.wso2.carbon.identity.api.server.media.service/org.wso2.carbon.identity.api.server.media.service.v1/src/main/java/org/wso2/carbon/identity/api/server/media/service/v1/core/ServerMediaService.java

```diff
@@ -0,0 +1,126 @@
+import javax.ws.rs.core.Response;
+
+/**
+ * Perform media service related operations.
+ */
+public class ServerMediaService {
+
+    private static final Log LOG = LogFactory.getLog(ServerMediaService.class);
+
+    /**
+     * Upload media.
+     *
+     * @param filesInputStream      The list of files to be uploaded as input streams.
+     * @param filesDetail           File details of the list of files to be uploaded.
+     * @param resourceFilesMetadata Metadata associated with the file upload.
+     * @param onlyEndUserAccess     If the resource is end-user protected.
+     * @return unique identifier of the uploaded media.
+     */
+    public String uploadMedia(List<InputStream> filesInputStream, List<Attachment> filesDetail,
+                              ResourceFilesMetadata resourceFilesMetadata, boolean onlyEndUserAccess) {
+
+        MediaMetadata mediaMetadata = new MediaMetadata();
+        mediaMetadata.setFileName(filesDetail.get(0).getContentDisposition().getFilename());
+        mediaMetadata.setFileContentType(filesDetail.get(0).getContentType().toString());
+
+        if (resourceFilesMetadata != null) {
+            mediaMetadata.setFileTag(resourceFilesMetadata.getFileTag());
+            FileSecurity fileSecurity;
+            if (onlyEndUserAccess) {
+                ArrayList<String> users = new ArrayList<>();
+                users.add(ContextLoader.getUsernameFromContext());
+                fileSecurity = new FileSecurity(false, users, null);
+            } else {
+                ResourceFilesMetadataFileSecurity fileSecurityMeta = resourceFilesMetadata.getFileSecurity();
+                fileSecurity = new FileSecurity(fileSecurityMeta.getAllowedAll(),
+                        fileSecurityMeta.getAllowedUsers(), fileSecurityMeta.getAllowedScopes());
+            }
+            mediaMetadata.setFileSecurity(fileSecurity);
+        }
+
+        StorageSystemManager storageSystemManager = MediaServiceDataHolder.getStorageSystemManager();
+        String tenantDomain = ContextLoader.getTenantDomainFromContext();
+        try {
+            return storageSystemManager.addFile(filesInputStream, mediaMetadata, tenantDomain);
+        } catch (StorageSystemException e) {
+            MediaServiceConstants.ErrorMessage errorMessage = MediaServiceConstants.ErrorMessage.
+                    ERROR_CODE_ERROR_UPLOADING_MEDIA;
+            Response.Status status = Response.Status.INTERNAL_SERVER_ERROR;
+            throw handleException(e, errorMessage, status);
+        }
+    }
+
+    private APIError handleException(Exception e, MediaServiceConstants.ErrorMessage errorEnum,
+                                     Response.Status status) {
+
+        ErrorResponse errorResponse = getErrorBuilder(errorEnum).build(LOG, e, errorEnum.getDescription());
+        return new APIError(status, errorResponse);
+    }
+
+    private APIError handleBadRequestException(MediaServiceConstants.ErrorMessage error, String data) {
```

Comment:
no need to explicitly handle bad requests.. you can map the error code to required http status code from here https://github.com/wso2/identity-rest-dispatcher/blob/master/components/org.wso2.carbon.identity.api.dispatcher/src/main/resources/ErrorMappings.properties